src/api.c

```diff
@@ -1534,8 +1539,55 @@ int cgroup_build_tasks_procs_path(char * const path,
 		err = 0;
 		break;
 	case CGROUP_V2:
-		strncat(path, "cgroup.procs", path_sz - strlen(path));
 		err = 0;
+		snprintf(cg_type_path, FILENAME_MAX, "%scgroup.type", path);
+
+		/* file cgroup.type, doesn't exist for root cgroup. */
+		stat_ret = stat(cg_type_path, &st);
+		if (stat_ret != 0) {
+			strncat(path, "cgroup.procs", path_sz - strlen(path));
+			break;
+		}
+
+		fp = fopen(cg_type_path, "re");
+		if (!fp) {
+			cgroup_warn("Warning: failed to open file %s: %s\n",
+					cg_type_path, strerror(errno));
+			err = ECGOTHER;
+			break;
+		}
+
+		if (fgets(cg_type, LL_MAX, fp) == NULL) {
+			fclose(fp);
+			cgroup_warn("Warning: failed to read file %s: %s\n",
+					cg_type_path, strerror(errno));
+			err = ECGOTHER;
+			break;
+		}
+
+		len = strlen(cg_type) - 1;
+		cg_type[len] = '\0';
+
+		/*
+		 * Append cgroup.threads to the path, if the cgroup.type is
+		 * threaded and cgroup.procs for type domain, domain threaded.
+		 * domain type is used for regular cgroup and domain threaded
+		 * for root of threaded cgroup v2 subtree.  Another possible
+		 * type is domain invalid, it's an invalid state, under the
+		 * threaded subtree.
+		 */
+		if (strcmp(cg_type, "domain") == 0 ||
+		    strcmp(cg_type, "domain threaded") == 0) {
```

Comment:
So i really do not like using strcmp here - I would rather you use a strncmp -> esp since the string we are compaing against has a fixed size?

Reply:
Sure, the reason to use `strcmp()` was to compare the whole string, that is read from `/proc` and there is very little chance of overrun.